packages/backend/src/open_payments/grantReference/service.ts

```diff
@@ -0,0 +1,44 @@
+import { Transaction, TransactionOrKnex } from 'objection'
+import { GrantReference } from './model'
+
+export interface GrantReferenceService {
+  get(grantId: string, trx?: Transaction): Promise<GrantReference>
+  create(
+    options: CreateGrantReferenceOptions,
+    trx?: Transaction
+  ): Promise<GrantReference>
+  lock(grantId: string, trx: TransactionOrKnex): Promise<void>
```

Comment:
```suggestion
  lock(grantId: string, trx: Transaction): Promise<void>
```

Reply:
This has implications on the `validateGrant` function in the `OutgoingPaymentService`. Right now, one of its input parameters is `deps:ServiceDependencies`, which includes `deps.knex:TransactionOrKnex`. If we don't want to use that, I have to specifically pass the transaction to `validateGrant`. Do you think we should make that change?

Reply:
- https://github.com/interledger/rafiki/issues/618